### PR TITLE
Handle custom attributes received in the API response.

### PR DESCRIPTION
**Pull request by wRAR**

### CODE REVIEW

**Comment by wRAR** [comment]
Requires https://github.com/zytedata/zyte-common-items/pull/100

**Comment by wRAR** [comment]
(some test failures related to field overrides)

**Review by Gallaecio — APPROVED** [review]

**Comment by Gallaecio** [comment]
I’ll handle the update of https://github.com/scrapy-plugins/scrapy-zyte-api/blob/a1d81d11854b420248f38e7db49c685a8d46d943/scrapy_zyte_api/_params.py#L25 since I am already refactoring it significantly as part of SERP work.

File: scrapy_zyte_api/providers.py [hunk 1]
```diff
@@ -186,6 +191,17 @@ async def __call__(  # noqa: C901
                         }
                     )
                 continue
+            if cls_stripped in {CustomAttributes, CustomAttributesValues}:
+                zyte_api_meta["customAttributes"] = {
+                    k: (
+                        dict(v)
+                        if isinstance(v, frozenset)
+                        else list(v) if isinstance(v, tuple) else v
+                    )
```
Comment (kmike) [inline scrapy_zyte_api/providers.py:200]: this code seems to be copy-pasted - time to add a helper function? Is it something like undo_make_hashable?

Comment (wRAR) [reply scrapy_zyte_api/providers.py:200]: Yeah.

Comment (wRAR) [reply scrapy_zyte_api/providers.py]: Done, and now I think both functions need tests. We don't actually test what we receive in mockserver requests.

Comment (kmike) [reply scrapy_zyte_api/providers.py]: Makes sense, let's add a few tests!

**Review by kmike — APPROVED** [review]
The code looks good. +1 to merge after CI is fixed, and some very basic tests are added for the hash/unhash functions.

**Comment by wRAR** [comment]
We should release zyte-common-items before merging this I think.

**Comment by kmike** [comment]
@wRAR feel free to merge when you think it's ready.

**Pull request merged**